PulseSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java

```diff
@@ -0,0 +1,205 @@
+import com.ooyala.android.skin.OoyalaSkinLayoutController;
+import com.ooyala.android.util.SDCardLogcatOoyalaEventsLogger;
+import com.ooyala.sample.utils.VideoItem;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+/**
+ * This class asks permission for WRITE_EXTERNAL_STORAGE. We need it for automation hooks
+ * as we need to write into the SD card and automation will parse this file.
+ */
+public abstract class AbstractHookActivity extends Activity implements Observer, DefaultHardwareBackBtnHandler {
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	String TAG = this.getClass().toString();
+	final int CLICK_THROUGH_REQUEST = 11234;
+	protected OoyalaSkinLayoutController playerSkinLayoutController;
+
+
+	SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+
+	String embedCode;
+	final String PCODE = "tlM2k6i2-WrXX1DE_b8zfhui_eQN";
+	final String DOMAIN = "http://ooyala.com";
+
+	OoyalaPlayer player;
+
+	boolean writePermission = false;
+	boolean asked = false;
+
+	// complete player setup after we asked for permission to write into external storage
+	abstract void completePlayerSetup(final boolean asked);
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+
+		if (ContextCompat.checkSelfPermission(this, WRITE_EXTERNAL_STORAGE) != PERMISSION_GRANTED) {
+			ActivityCompat.requestPermissions(this, new String[]{WRITE_EXTERNAL_STORAGE}, PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE);
+		} else {
+			writePermission= true;
+			asked = true;
+		}
+		embedCode = getIntent().getExtras().getString("embed_code");
+	}
+
+	@Override
+	public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+		if (requestCode == PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE) {
+			asked = true;
+			if (grantResults.length > 0 && grantResults[0] == PERMISSION_GRANTED) {
+				writePermission = true;
+			}
+			completePlayerSetup(asked);
+		}
+	}
+
+	@Override
+	protected void onStop() {
+		super.onStop();
+		Log.d(TAG, "Player Activity Stopped");
+		if (null != player) {
+			player.suspend();
+		}
+	}
+
+	@Override
+	protected void onRestart() {
+		super.onRestart();
+		Log.d(TAG, "Player Activity Restarted");
+		if (null != player) {
+			player.resume();
+		}
+	}
+
+	@Override
+	public boolean onKeyDown(int keyCode, KeyEvent event) {
+		playerSkinLayoutController.onKeyDown(keyCode, event);
+		return super.onKeyDown(keyCode, event);
+	}
+
+	/** Start DefaultHardwareBackBtnHandler **/
+	@Override
+	public void invokeDefaultOnBackPressed() {
+		super.onBackPressed();
+	}
+	/** End DefaultHardwareBackBtnHandler **/
+
+	/** Start Activity methods for Skin **/
+	@Override
+	protected void onPause() {
+		super.onPause();
+		if (playerSkinLayoutController != null) {
```

Comment:
when I asked to change null, i meant for all cases when you encounter them, you just changed it for when you ask for the player. Whenever you check for null use it as `null != variable`